queue-event-handler.js

```diff
@@ -1,18 +1,40 @@
 const postal = require('postal');
 const Github = require('./github');
 const logger = require('./logger');
-const { tail } = require('ramda');
+const { slice } = require('ramda');
 const pullRequestManager = require('./manager-pullrequest');
 const queueManager = require('./manager-queue');
 const { unpackQueueName } = require('./helpers/queue-helpers');
 
-const addItem = ({ queue }) => {
+const MAX_EXPLICIT_QUEUE_LENGTH = 3;
+
+const addItem = ({ queue, item, index }) => {
     const { owner, repo, branch } = unpackQueueName(queue);
+    const { pullRequestNumber: itemPullRequestNumber } = item;
+
+    const blockNewPullRequestIfNecessary = (index > MAX_EXPLICIT_QUEUE_LENGTH)
```

Comment:
Are we sure we wanna block PRs falling over the limit? We can even ignore them at all. What do you think?

Reply:
I think it's important that we do, otherwise after booking the PR will still say "Book to merge" which may be confusing.

Reply:
If we ignore them it'll just say _Waiting for status to be reported_